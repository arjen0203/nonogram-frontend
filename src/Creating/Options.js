import React, {Component} from 'react';

class Options extends Component {
    render() {
        return (
            <div className="center">
                <div data-testid="options" className="options">
                    <label htmlFor="name">Name: </label>
                    <input data-testid="optName" id="name" className="name-input" type="text" value={this.props.name} onChange={(e) => this.props.changeName(e.target.value)}></input>
                    <div className="name-warning">{this.props.nameWarning}</div>
                    <div>
                        <label htmlFor="width">Width:</label>
                        <input data-testid="optHeight" id="width" className="size-input" type="number" min="1" max="30"
                               value={this.props.height} onChange={(e) => this.props.changeHeight(e.target.value)}></input>
                        <label htmlFor="height">Height:</label>
                        <input data-testid="optWidth" id="height" className="size-input" type="number" min="1" max="30"
                               value={this.props.width} onChange={(e) => this.props.changeWidth(e.target.value)}></input>
                    </div>
                    <div className="size-warning">{this.props.sizeWarning}</div>
                    <button onClick={() => this.props.saveNonogram()} className="save-button">Save nonogram</button>
                    <div className="save-state">{this.props.saveState}</div>
                </div>
            </div>
        );
    }
}

export default Options;