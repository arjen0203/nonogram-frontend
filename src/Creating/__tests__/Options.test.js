import React from "react";
import ReactDOM from 'react-dom';
import Options from "../Options";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<Options></Options>, div);
});

it('Render correctly empty', function () {
	const rendered = render(<Options width="5" height="6" name="rens"></Options>)
	const compHeight = rendered.getByTestId("optHeight");
	const compWidth = rendered.getByTestId("optWidth");
	const compName = rendered.getByTestId("optName");

	expect(compHeight).toHaveValue(6);
	expect(compWidth).toHaveValue(5);
	expect(compName).toHaveValue("rens");
});

it('Render correctly empty', function () {
	const rendered = render(<Options width="8" height="4" name="sjoerd"></Options>)
	const compHeight = rendered.getByTestId("optHeight");
	const compWidth = rendered.getByTestId("optWidth");
	const compName = rendered.getByTestId("optName");

	expect(compHeight).toHaveValue(4);
	expect(compWidth).toHaveValue(8);
	expect(compName).toHaveValue("sjoerd");
});

it('matches snapshot', function () {
	const tree = renderer.create(<Options></Options>).toJSON();
	expect(tree).toMatchSnapshot();
});