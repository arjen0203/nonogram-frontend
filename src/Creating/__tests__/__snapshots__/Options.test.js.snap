// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="center"
>
  <div
    className="options"
    data-testid="options"
  >
    <label
      htmlFor="name"
    >
      Name: 
    </label>
    <input
      className="name-input"
      data-testid="optName"
      id="name"
      onChange={[Function]}
      type="text"
    />
    <div
      className="name-warning"
    />
    <div>
      <label
        htmlFor="width"
      >
        Width:
      </label>
      <input
        className="size-input"
        data-testid="optHeight"
        id="width"
        max="30"
        min="1"
        onChange={[Function]}
        type="number"
      />
      <label
        htmlFor="height"
      >
        Height:
      </label>
      <input
        className="size-input"
        data-testid="optWidth"
        id="height"
        max="30"
        min="1"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="size-warning"
    />
    <button
      className="save-button"
      onClick={[Function]}
    >
      Save nonogram
    </button>
    <div
      className="save-state"
    />
  </div>
</div>
`;
